packages/web-components/src/components/menu/menu.ts

```diff
@@ -0,0 +1,516 @@
+   */
+  @property({ type: Boolean })
+  isRoot = true;
+
+  /**
+   * Document direction.
+   */
+  @property({ type: String })
+  direction = 'ltr';
+  /**
+   * Open value for the menu .
+   */
+  @property({ type: Boolean })
+  open = true;
+  /**
+   * Active element in the DOM .
+   */
+  @property({ type: HTMLElement })
+  focusreturn;
+  /**
+   * Position of the Menu .
+   */
+  @property()
+  position = [-1, -1];
+  /**
+   * Size attribute .
+   */
+  @property({ attribute: true })
+  size = MENU_SIZE.SMALL;
+  /**
+   * Deprecated: Menus now always support both icons as well as selectable items and nesting. The mode of this menu. Defaults to full. full supports nesting and selectable menu items, but no icons. basic supports icons but no nesting or selectable menu items.
+
+    This prop is not intended for use and will be set by the respective implementation (like useContextMenu, MenuButton, and ComboButton).
+   */
+  @property()
+  mode;
+
+  /**
+   * Specify how the menu should align with the button element
+   */
+  @property({ type: String })
+  menuAlignment;
+  /**
+   * Position of the Menu in X axis .
+   */
+  @property()
+  x: number | number[] = 0;
+  /**
+   * Position of the Menu in Y axis .
+   */
+  @property()
+  y: number | number[] = 0;
+
+  /**
+   * The name of the custom event fired when the the Menu should be closed.
+   */
+  static get eventOnClose() {
+    return `${prefix}-menu-closed`;
+  }
+  /**
+   * The name of the custom event fired when the the Menu should be opened.
+   */
+  static get eventOnOpen() {
+    return `${prefix}-menu-opened`;
+  }
+  updated(changedProperties) {
+    if (changedProperties.has('open') && this.open) {
+      this._handleOpen();
+    }
+  }
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener('icon-detect', () => {
+      this.shadowRoot
+        ?.querySelector('.cds--menu')
+        ?.classList.add(`${prefix}--menu--with-icons`);
+    });
+  }
+  async firstUpdated() {
+    this.isRtl = this.direction === 'rtl';
+    this.isRoot = this.context.isRoot;
+
+    if (this.isChild) {
+      this._newContextCreate();
+    }
+    await this.updateComplete;
+    this._registerMenuItems();
+    this._setActiveItems();
+  }
+  render() {
+    const {
+      open,
+      menuAlignment,
+      label,
+      position,
+      _handleKeyDown: handleKeyDown,
+    } = this;
+    const menuClasses = classMap({
+      [`${prefix}--menu`]: true,
+      [`${prefix}--menu--${this.size}`]: this.size,
+      [`${prefix}--menu--box-shadow-top`]:
+        menuAlignment && menuAlignment.slice(0, 3) === 'top',
+      [`${prefix}--menu--open`]: open,
+      [`${prefix}--menu--shown`]: position[0] >= 0 && position[1] >= 0,
+      [`${prefix}--menu--with-selectable-items`]:
+        this.context.hasSelectableItems,
+    });
+    return html`
+      <ul
+        class="${menuClasses}"
+        aria-label="${label}"
+        tabindex="1"
```

Comment:
Double check the accessibility checker. When looking I saw this and compared to react and it looks like it should be set to "-1"
```suggestion
        tabindex="-1"
```